includes/commerce_pos.transaction.inc

```diff
@@ -142,6 +142,36 @@ function commerce_pos_transaction_form(&$form, &$form_state, $transaction_type)
     unset($form_state['lookup_result']);
   }
 
+  // Let's add a order search textfield if we're in a return transaction so the
+  // user could easily pick out which items from the order needs to be returned.
+  if ($return) {
+    $form['order_search'] = array(
+      '#type' => 'container',
+      '#prefix' => '<div id="commerce-pos-return-order-search">',
+      '#tree' => TRUE,
+      '#suffix' => '</div>',
+    );
+    $form['order_search']['input'] = array(
+      '#type' => 'textfield',
+      '#title' => '',
+      '#size' => 60,
+      '#description' => t('Search by order ID to load the items in an order.<br>(Note: This does not support searching for Order Number.)'),
+      '#attributes' => array(
+        'placeholder' => t('Order Search'),
+      ),
+      '#default_value' => !empty($form_state['values']['order_search']['input']) ? $form_state['values']['order_search']['input'] : NULL,
+    );
+
+    $form['order_search']['search'] = array(
+      '#type' => 'button',
+      '#value' => t('Go'),
```

Comment:
I don't think "Go" is the right word here; we can probably find something that describes the button's function. What about something along the lines of "Returns items from this order"?

Eg, see Clear Labels here https://cadmium.acromedia.com/#g=1&p=buttons